From 9f70b09d7df45c22183035370e09a6bd87753dd3 Mon Sep 17 00:00:00 2001
From: Elizabeth Liner <eliner@us.ibm.com>
Date: Mon, 14 Nov 2016 17:07:50 -0600
Subject: [PATCH] Makefile changes for OP Build CI Witherspoon

Change-Id: Id51c5333c10e910f54d9f2fb5666ed5701efbfc0
---

diff --git a/src/usr/testcore/lib/makefile b/src/usr/testcore/lib/makefile
index 9866637..a688b2a 100644
--- a/src/usr/testcore/lib/makefile
+++ b/src/usr/testcore/lib/makefile
@@ -25,8 +25,8 @@
 ROOTPATH = ../../../..
 
 MODULE = testsyslib
-//  @TODO-RTC:151185-Turn enable all test cases
-// TESTS = *.H
+#@TODO-RTC:151185-Turn enable all test cases
+#TESTS = *.H
 TESTS = stltest.H
 
 
