diff --git a/platforms/astbmc/vesnin.c b/platforms/astbmc/vesnin.c
index 39c5ed87..bdf6fca1 100644
--- a/platforms/astbmc/vesnin.c
+++ b/platforms/astbmc/vesnin.c
@@ -35,8 +35,11 @@
 /* IPMI message identifier (IBM OEM) for PCI inventory. */
 #define IPMI_PCI_INV IPMI_CODE(0x3a, 0x2a)
 
+//#define VESNIN_PCI_INVENTORY
+#ifdef VESNIN_PCI_INVENTORY
 /* Id of the current PCI inventory synchronization session. */
 static uint8_t pci_inv_session_id;
+#endif // VESNIN_PCI_INVENTORY
 
 /**
  * struct pciinv_device - PCI device inventory description.
@@ -262,6 +265,7 @@ static const struct slot_table_entry vesnin_phb_table[] = {
 	{ .etype = st_end }
 };
 
+#ifdef VESNIN_PCI_INVENTORY
 /**
  * pciinv_walk() - Callback from PCI enumerator, see :c:func:`pci_walk_dev`.
  */
@@ -300,18 +304,23 @@ static int pciinv_walk(struct phb *phb, struct pci_device *pd, void *data __unus
 
 	return OPAL_SUCCESS;
 }
+#endif // VESNIN_PCI_INVENTORY
 
 static void vesnin_pci_probe_complete(void)
 {
+#ifdef VESNIN_PCI_INVENTORY
 	struct phb *phb;
+#endif // VESNIN_PCI_INVENTORY
 
 	check_all_slot_table();
 
+#ifdef VESNIN_PCI_INVENTORY
 	/* Send PCI device list to the BMC */
 	++pci_inv_session_id;
 	for_each_phb(phb) {
 		pci_walk_dev(phb, NULL, &pciinv_walk, NULL);
 	}
+#endif // VESNIN_PCI_INVENTORY
 }
 
 static bool vesnin_probe(void)
