From 50ae9cf95223a5f6492e7e515ccef9d4499ef823 Mon Sep 17 00:00:00 2001
From: Jim Yuan <jim.yuan@supermicro.com>
Date: Tue, 11 Oct 2016 10:03:01 -0700
Subject: [PATCH 05/14] change pnor response command to 0x7

Signed-off-by: Jim Yuan <jim.yuan@supermicro.com>
---
 src/include/usr/ipmi/ipmiif.H | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/include/usr/ipmi/ipmiif.H b/src/include/usr/ipmi/ipmiif.H
index 94fca5fb6..4b41cb3bc 100644
--- a/src/include/usr/ipmi/ipmiif.H
+++ b/src/include/usr/ipmi/ipmiif.H
@@ -289,7 +289,7 @@ namespace IPMI
     { return std::make_pair(NETFUN_IBM, 0x07); }
 
     inline const command_t pnor_response(void)
-    { return std::make_pair(NETFUN_IBM, 0x08); }
+	{ return std::make_pair(NETFUN_IBM, 0x07); }
 
     inline const command_t hiomap_event(void)
     { return std::make_pair(NETFUN_IBM, 0x0f); }
-- 
2.16.2.windows.1

