From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joel Stanley <joel@jms.id.au>
Date: Mon, 3 Dec 2018 10:10:31 +1030
Subject: [PATCH 2/2] Release OpenPower kernel

Signed-off-by: Joel Stanley <joel@jms.id.au>
---
 Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 3b9e4658d31f..d836f4d1787c 100644
--- a/Makefile
+++ b/Makefile
@@ -2,8 +2,9 @@
 VERSION = 4
 PATCHLEVEL = 20
 SUBLEVEL = 3
-EXTRAVERSION =
+EXTRAVERSION = -openpower1
 NAME = Shy Crocodile
+NAME = "People's Front"
 
 # *DOCUMENTATION*
 # To see a list of typical targets execute "make help"
