From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joel Stanley <joel@jms.id.au>
Date: Wed, 27 Mar 2019 12:53:02 +1030
Subject: [PATCH 2/2] Release OpenPower kernel

Signed-off-by: Joel Stanley <joel@jms.id.au>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 06fda21614bc..8022d6a2faed 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 5
 PATCHLEVEL = 0
 SUBLEVEL = 4
-EXTRAVERSION =
+EXTRAVERSION = -openpower1
 NAME = Shy Crocodile
 
 # *DOCUMENTATION*
