From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joel Stanley <joel@jms.id.au>
Date: Mon, 8 Apr 2019 16:42:01 +0930
Subject: [PATCH 3/3] Release OpenPower kernel

Signed-off-by: Joel Stanley <joel@jms.id.au>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index af99c77c7066..cd31b792f27c 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 5
 PATCHLEVEL = 0
 SUBLEVEL = 7
-EXTRAVERSION =
+EXTRAVERSION = -openpower1
 NAME = Shy Crocodile
 
 # *DOCUMENTATION*
