From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joel Stanley <joel@jms.id.au>
Date: Thu, 31 May 2018 01:27:32 +0930
Subject: [PATCH 3/3] Release OpenPower kernel

Signed-off-by: Joel Stanley <joel@jms.id.au>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 146e527a5e06..30edc0ce365f 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 4
 PATCHLEVEL = 16
 SUBLEVEL = 13
-EXTRAVERSION =
+EXTRAVERSION = -openpower1
 NAME = Fearless Coyote
 
 # *DOCUMENTATION*
