From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Stewart Smith <stewart@linux.ibm.com>
Date: Fri, 11 May 2018 17:41:12 -0700
Subject: [PATCH 3/3] Release OpenPower kernel

Signed-off-by: Stewart Smith <stewart@linux.ibm.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 1c5d5d8c45e2..ea0dccd03e90 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 4
 PATCHLEVEL = 16
 SUBLEVEL = 7
-EXTRAVERSION =
+EXTRAVERSION = -openpower2
 NAME = Fearless Coyote
 
 # *DOCUMENTATION*
