From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joel Stanley <joel@jms.id.au>
Date: Mon, 4 Jun 2018 17:14:35 +0930
Subject: [PATCH 3/3] Release OpenPower kernel

Signed-off-by: Joel Stanley <joel@jms.id.au>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index e551c9af6a06..1ec18f8a0e72 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 4
 PATCHLEVEL = 17
 SUBLEVEL = 1
-EXTRAVERSION =
+EXTRAVERSION = -openpower1
 NAME = Merciless Moray
 
 # *DOCUMENTATION*
