From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joel Stanley <joel@jms.id.au>
Date: Mon, 26 Feb 2018 10:29:07 +1030
Subject: [PATCH 4/4] Release OpenPower kernel

Signed-off-by: Joel Stanley <joel@jms.id.au>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 38acc6047d7d..5d52a43d2258 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 4
 PATCHLEVEL = 14
 SUBLEVEL = 24
-EXTRAVERSION =
+EXTRAVERSION = -openpower1
 NAME = Petit Gorille
 
 # *DOCUMENTATION*
