From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Samuel Mendoza-Jonas <sam@mendozajonas.com>
Date: Tue, 17 Oct 2017 13:32:56 +1100
Subject: [PATCH 4/4] Release OpenPower kernel

Signed-off-by: Samuel Mendoza-Jonas <sam@mendozajonas.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 0d4f1b19869d..ad0f7fd78596 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 13
 SUBLEVEL = 7
-EXTRAVERSION =
+EXTRAVERSION = -openpower1
 NAME = Fearless Coyote
 
 # *DOCUMENTATION*
