From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joel Stanley <joel@jms.id.au>
Date: Wed, 16 May 2018 13:42:40 +0930
Subject: [PATCH 4/4] Release OpenPower kernel

Signed-off-by: Joel Stanley <joel@jms.id.au>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 5da6ffd69209..6d6fbae5e2b9 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 4
 PATCHLEVEL = 16
 SUBLEVEL = 8
-EXTRAVERSION =
+EXTRAVERSION = -openpower2
 NAME = Fearless Coyote
 
 # *DOCUMENTATION*
