From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joel Stanley <joel@jms.id.au>
Date: Thu, 29 Mar 2018 17:10:52 +1100
Subject: [PATCH 4/4] Release OpenPower kernel

Signed-off-by: Joel Stanley <joel@jms.id.au>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index a5e561900daf..3c8e12e69a5b 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 4
 PATCHLEVEL = 15
 SUBLEVEL = 14
-EXTRAVERSION =
+EXTRAVERSION = -openpower1
 NAME = Fearless Coyote
 
 # *DOCUMENTATION*
