From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Samuel Mendoza-Jonas <sam@mendozajonas.com>
Date: Mon, 23 Oct 2017 12:57:15 +1100
Subject: [PATCH 5/5] Release OpenPower kernel

Signed-off-by: Samuel Mendoza-Jonas <sam@mendozajonas.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 66ec023da822..4b8c4cb589ac 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 13
 SUBLEVEL = 8
-EXTRAVERSION =
+EXTRAVERSION = -openpower2
 NAME = Fearless Coyote
 
 # *DOCUMENTATION*
