From 3f55ad27d67e45d47cea521600f6bd76cc021704 Mon Sep 17 00:00:00 2001
From: Joel Stanley <joel@jms.id.au>
Date: Tue, 16 Aug 2016 23:00:23 +0930
Subject: [PATCH 9/9] Release 4.4.31-openpower1

Signed-off-by: Joel Stanley <joel@jms.id.au>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 7c6f28e7a2f6..9bf5c3530b98 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 4
 SUBLEVEL = 31
-EXTRAVERSION =
+EXTRAVERSION = -openpower1
 NAME = Blurry Fish Butt
 
 # *DOCUMENTATION*
-- 
2.10.2

