From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joel Stanley <joel@jms.id.au>
Date: Wed, 7 Jun 2017 14:46:38 +0930
Subject: [PATCH 11/11] Release v4.4.77-openpower1

Signed-off-by: Joel Stanley <joel@jms.id.au>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index bf49a61d02e2..0bcf47141d7c 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 4
 SUBLEVEL = 77
-EXTRAVERSION =
+EXTRAVERSION = -openpower1
 NAME = Blurry Fish Butt
 
 # *DOCUMENTATION*
