From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joel Stanley <joel@jms.id.au>
Date: Tue, 28 Feb 2017 12:15:10 +1030
Subject: [PATCH 15/15] Release OpenPower kernel

Signed-off-by: Joel Stanley <joel@jms.id.au>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 09eccff4f569..e291929b4fd0 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 10
 SUBLEVEL = 1
-EXTRAVERSION =
+EXTRAVERSION = -openpower1
 NAME = Fearless Coyote
 
 # *DOCUMENTATION*
-- 
2.11.0

