From bf1e944d54077e1452092898069a0dacf18c381a Mon Sep 17 00:00:00 2001
From: Joel Stanley <joel@jms.id.au>
Date: Tue, 20 Oct 2015 15:01:06 +1030
Subject: [PATCH 17/17] Release 4.2.4-openpower1

Signed-off-by: Joel Stanley <joel@jms.id.au>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index a952801..d8175be 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 2
 SUBLEVEL = 4
-EXTRAVERSION =
+EXTRAVERSION = -openpower1
 NAME = Hurr durr I'ma sheep
 
 # *DOCUMENTATION*
-- 
2.5.0

