From 5c51532db6ffcf1ec521a8a0d19cf3e814f6b428 Mon Sep 17 00:00:00 2001
From: Joel Stanley <joel@jms.id.au>
Date: Fri, 11 Sep 2015 16:09:56 +0930
Subject: [PATCH 18/18] Release 4.2.1-openpower1

Signed-off-by: Joel Stanley <joel@jms.id.au>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index a03efc1..49bbfb6 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 2
 SUBLEVEL = 1
-EXTRAVERSION =
+EXTRAVERSION = -openpower1
 NAME = Hurr durr I'ma sheep
 
 # *DOCUMENTATION*
-- 
2.5.0

