From 185b564ab788e84bb7b07482f7fdbb816b209b46 Mon Sep 17 00:00:00 2001
From: Stewart Smith <stewart@linux.vnet.ibm.com>
Date: Sat, 3 Oct 2015 11:07:20 +1000
Subject: [PATCH 2/2] Release 4.2.1-openpower2

Signed-off-by: Stewart Smith <stewart@linux.vnet.ibm.com>
---
 Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index a03efc1..ca21a06 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 2
 SUBLEVEL = 1
-EXTRAVERSION =
+EXTRAVERSION = -openpower2
 NAME = Hurr durr I'ma sheep
 
 # *DOCUMENTATION*
-- 
1.7.1

