From da9578199b0248e0f0b7b72318dd7275f4044767 Mon Sep 17 00:00:00 2001
From: Patrick Williams <iawillia@us.ibm.com>
Date: Thu, 11 Dec 2014 17:36:58 -0600
Subject: [PATCH] Fix compile fail in tracepp.

---
 src/tracepp/tracepp.C | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/tracepp/tracepp.C b/src/tracepp/tracepp.C
index 8d5d911..a197dd5 100755
--- a/src/tracepp/tracepp.C
+++ b/src/tracepp/tracepp.C
@@ -50,6 +50,7 @@
 #include <fcntl.h>
 #include <map>
 #include <vector>
+#include <unistd.h>
 
 #include <sys/types.h>
 typedef u_int32_t u32 ;
-- 
2.1.1

