From 4c5f6ff7a2b7c2df17d16c3ce18b19036be95223 Mon Sep 17 00:00:00 2001
From: Jim Yuan <jim.yuan@supermicro.com>
Date: Mon, 17 Dec 2018 15:57:05 -0800
Subject: [PATCH] VGA handshake with BMC

---
 ast_main.c | 37 +++++++++++++++++++++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/drivers/gpu/drm/ast/ast_main.c b/drivers/gpu/drm/ast/ast_main.c
index f167cdf..1d46961 100644
--- a/drivers/gpu/drm/ast/ast_main.c
+++ b/drivers/gpu/drm/ast/ast_main.c
@@ -478,6 +478,39 @@ static u32 ast_get_vram_info(struct drm_device *dev)
 	return vram_size;
 }
 
+static int ast_bmc_handshake(struct ast_private *ast) {
+	const u8 h2s_addr = 0x9a, s2h_addr = 0xd2;
+	const int handshake_timeout = 10000;
+	u8 soc_reg, last_reg;
+	int retries;
+
+	soc_reg = ast_get_index_reg(ast, AST_IO_CRTC_PORT, s2h_addr);
+	pr_info("%s: soc reg: %02x\n", __func__, soc_reg);
+	if (!(soc_reg & 0x1))
+		return 0;
+
+	last_reg = soc_reg;
+
+	ast_set_index_reg(ast, AST_IO_CRTC_PORT, h2s_addr, 0x01);
+
+	for (retries = 0; retries < handshake_timeout; retries++) {
+		soc_reg = ast_get_index_reg(ast, AST_IO_CRTC_PORT, s2h_addr);
+		if (soc_reg != last_reg) {
+			pr_info("%s: change %02x\n", __func__, soc_reg);
+			last_reg = soc_reg;
+		}
+		if (!(soc_reg & 0x1)){
+		    ast_set_index_reg(ast, AST_IO_CRTC_PORT, h2s_addr, 0x00);
+			return 0;
+		}
+
+		msleep(10);
+	}
+
+	DRM_ERROR("Failed to complete BMC handshake");
+	return -EIO;
+}
+
 int ast_driver_load(struct drm_device *dev, unsigned long flags)
 {
 	struct ast_private *ast;
@@ -518,6 +551,10 @@ int ast_driver_load(struct drm_device *dev, unsigned long flags)
 
 	ast_detect_chip(dev, &need_post);
 
+	ret = ast_bmc_handshake(ast);
+	if (ret)
+		goto out_free;
+
 	if (need_post)
 		ast_post_gpu(dev);
 
-- 
2.16.2.windows.1

