diff --git a/platforms/astbmc/witherspoon.c b/platforms/astbmc/witherspoon.c
index cb09eefe048c..85336c626816 100644
--- a/platforms/astbmc/witherspoon.c
+++ b/platforms/astbmc/witherspoon.c
@@ -64,6 +64,22 @@ static bool witherspoon_probe(void)
 	return true;
 }
 
+static bool stonewither_probe(void)
+{
+	if (!dt_node_is_compatible(dt_root, "ibm,stonewither"))
+		return false;
+
+	/* Lot of common early inits here */
+	astbmc_early_init();
+
+	/* Setup UART for use by OPAL (Linux hvc) */
+	uart_set_console_policy(UART_CONSOLE_OPAL);
+
+	vpd_dt_fixup();
+
+	return true;
+}
+
 static void phb4_activate_shared_slot_witherspoon(struct proc_chip *chip)
 {
 	uint64_t val;
@@ -170,3 +186,19 @@ DECLARE_PLATFORM(witherspoon) = {
 
 	.pci_get_slot_info	= dt_slot_get_slot_info,
 };
+DECLARE_PLATFORM(stonewither) = {
+	.name			= "Stonewither",
+	.probe			= stonewither_probe,
+	.init			= astbmc_init,
+	.pre_pci_fixup		= witherspoon_pre_pci_fixup,
+	.start_preload_resource	= flash_start_preload_resource,
+	.resource_loaded	= flash_resource_loaded,
+	.bmc			= &astbmc_openbmc,
+	.cec_power_down         = astbmc_ipmi_power_down,
+	.cec_reboot             = astbmc_ipmi_reboot,
+	.elog_commit		= ipmi_elog_commit,
+	.exit			= ipmi_wdt_final_reset,
+	.terminate		= ipmi_terminate,
+
+	.pci_get_slot_info	= dt_slot_get_slot_info,
+};
