From 11bff93ad1ca613b1db9d96e9ea9c46b7d7bf48d Mon Sep 17 00:00:00 2001
From: Jeremy Kerr <jk@ozlabs.org>
Date: Thu, 5 Mar 2015 22:44:40 +1100
Subject: [PATCH] Makefile: specify abiv1

Signed-off-by: Jeremy Kerr <jk@ozlabs.org>
---
 Makefile.main | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile.main b/Makefile.main
index 3c67037..76f0bd3 100644
--- a/Makefile.main
+++ b/Makefile.main
@@ -50,7 +50,8 @@ CPPFLAGS += -DBITS_PER_LONG=64 -DHAVE_BIG_ENDIAN
 # causing all our printf's to warn
 CPPFLAGS += -ffreestanding
 
-CFLAGS := -fno-strict-aliasing -fstack-protector-all -pie -mbig-endian -m64
+CFLAGS := -fno-strict-aliasing -fstack-protector-all -pie \
+	  -mbig-endian -m64 -mabi=elfv1
 
 ifeq ($(STACK_CHECK),1)
 CFLAGS += -fstack-protector-all -pg
-- 
1.9.1

