From 11bff93ad1ca613b1db9d96e9ea9c46b7d7bf48d Mon Sep 17 00:00:00 2001
From: Jeremy Kerr <jk@ozlabs.org>
Date: Thu, 5 Mar 2015 22:44:40 +1100
Subject: [PATCH] Makefile: specify abiv1

Signed-off-by: Jeremy Kerr <jk@ozlabs.org>
---
 Makefile.main | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile.main b/Makefile.main
index 3c67037..76f0bd3 100644
--- a/Makefile.main
+++ b/Makefile.main
@@ -51,6 +51,7 @@ CPPFLAGS += -DBITS_PER_LONG=64 -DHAVE_BIG_ENDIAN
 CPPFLAGS += -ffreestanding
 
 CFLAGS := -fno-strict-aliasing -fstack-protector-all -pie -mbig-endian -m64
+CFLAGS += -mabi=elfv1
 CFLAGS += -Wl,--oformat,elf64-powerpc
 
 ifeq ($(SKIBOOT_GCOV),1)
-- 
1.9.1

