From ec49ccee122c1be9b0d90c0f810a8964575bd57f Mon Sep 17 00:00:00 2001
From: Jeremy Kerr <jk@ozlabs.org>
Date: Wed, 11 Feb 2015 15:47:17 +0800
Subject: [PATCH] Makefile: specify -mbig-endian in link flags

Signed-off-by: Jeremy Kerr <jk@ozlabs.org>
---
 Makefile.main | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.main b/Makefile.main
index b5376fc..0fca221 100644
--- a/Makefile.main
+++ b/Makefile.main
@@ -64,7 +64,7 @@ CFLAGS += $(CWARNS) $(OPTS) $(DBG)
 
 LDFLAGS := -m64 -static -nostdlib -Wl,--gc-sections -pie
 LDFLAGS += -Wl,-Ttext-segment,$(LD_TEXT) -Wl,-N -Wl,--build-id=none
-LDFLAGS += -Wl,--no-multi-toc
+LDFLAGS += -Wl,--no-multi-toc -mbig-endian -Wl,--oformat,elf64-powerpc
 LDRFLAGS=-melf64ppc
 # Debug stuff
 #LDFLAGS += -Wl,-v -Wl,-Map,foomap 
-- 
1.9.1

