From 3600e0e9bf8bb45039ba32df45994c28bb6419cc Mon Sep 17 00:00:00 2001
From: Patrick Williams <iawillia@us.ibm.com>
Date: Tue, 4 Aug 2015 11:27:09 -0500
Subject: [PATCH] Fix compile failures on RHEL6.

---
 Makefile.main | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/Makefile.main b/Makefile.main
index 3851c20..1f2b673 100644
--- a/Makefile.main
+++ b/Makefile.main
@@ -19,11 +19,9 @@ CWARNS := -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs \
 	  -Werror-implicit-function-declaration -Wdeclaration-after-statement \
 	  -Wno-pointer-sign -Wextra -Wno-sign-compare \
 	  -Wmissing-prototypes -Wmissing-declarations \
-	  -Wwrite-strings -Wcast-align -Wjump-misses-init \
+	  -Wwrite-strings -Wcast-align \
 	  -Winit-self \
-	  -Wsuggest-attribute=const \
-	  -Wsuggest-attribute=noreturn \
-	  -Wframe-larger-than=1024 -Wstack-usage=1024 \
+	  -Wframe-larger-than=1024 \
 	  -Werror -Wno-error=format
 
 # Host tools and options
-- 
2.4.4

